experiments/conductor/cmd/runner/controller_commands.go

```diff
@@ -464,6 +465,302 @@ func moveTestToSubDir(ctx context.Context, opts *RunnerOptions, branch Branch, e
+		strings.ToLower(branch.Kind),
+	)
+	currentYear := time.Now().Year()
+
+	// TODO: Customize the max attempt number.
+	for attempt := 0; attempt < 3; attempt++ {
+		log.Printf("Attempt %d to ensure full coverage", attempt+1)
+		// 1. Identify missing fields for target resource.
+		//
+		// Run TestCRDFieldPresenceInUnstructured for two times:
+		// First run: Optionally, update tests/apichecks/testdata/exceptions/missingfields.txt
+		// Second run: Verify there is no true errors.
+		cfg := CommandConfig{
+			Name: "API Checks: TestCRDFieldPresenceInUnstructured",
+			Cmd:  "go",
+			Args: []string{
+				"test", "-v", "-run", "TestCRDFieldPresenceInUnstructured",
+				filepath.Join("tests", "apichecks", "crds_test.go"),
+			},
+			WorkDir:     opts.branchRepoDir,
+			Env:         map[string]string{"WRITE_GOLDEN_OUTPUT": "1"},
+			MaxAttempts: 2,
+		}
+
+		_, err := executeCommand(opts, cfg)
+		if err != nil {
+			return nil, nil, fmt.Errorf("failed to run TestCRDFieldPresenceInUnstructured: %w", err)
+		}
+
+		missingFieldsFilePath := filepath.Join(
+			opts.branchRepoDir,
+			"tests", "apichecks", "testdata", "exceptions", "missingfields.txt",
+		)
+		crdName := fmt.Sprintf("%s.%s.cnrm.cloud.google.com", strings.ToLower(pluralOf(branch.Kind)), branch.Group)
+		prefix := fmt.Sprintf("[missing_field] crd=%s version=%s", crdName, crdVersion)
+		lines, err := matchLineInFileByPrefix(missingFieldsFilePath, prefix)
+		if err != nil {
+			return nil, nil, fmt.Errorf("error matching prefix %q: %w", prefix, err)
+		}
+		if len(lines) == 0 && attempt > 0 {
+			log.Printf("No need to generate more tests for branch %s, all fields are covered", branch.Name)
+			break
+		}
+
+		missingFields, err := extractMissingFieldsFromExceptionLine(lines)
+		if err != nil {
+			return nil, nil, fmt.Errorf("error extracting missing fields: %w", err)
+		}
+
+		// TODO: Remove missing fields that are output-only.
+
+		// 2. Check if test case already exists, and create the directory if not.
+		testDir := filepath.Join(
+			parentDir,
+			fmt.Sprintf("%s-%s", strings.ToLower(branch.Kind), opts.testDirSuffix),
+		)
+
+		if attempt > 0 {
+			testDir = filepath.Join(fmt.Sprintf("%s-%d", testDir, attempt))
+		}
+		fullTestDir := filepath.Join(opts.branchRepoDir, testDir)
+		_, err = os.Stat(fullTestDir)
+		if err == nil {
+			// TODO: Add an option to not override the existing test.
+			log.Printf("Overriding test under %s", fullTestDir)
+		}
+		if err != nil && !errors.Is(err, os.ErrNotExist) {
+			return nil, nil, fmt.Errorf("error checking whether test directory %s exists: %w", fullTestDir, err)
+		} else {
+			if err := os.MkdirAll(fullTestDir, 0755); err != nil {
+				return nil, nil, fmt.Errorf("failed to create test directory %s: %w", fullTestDir, err)
+			}
+		}
+
+		// 3. Generate testdata.
+		// Use codebot to generate the initial create.yaml content.
+		createPrompt := fmt.Sprintf(`Generate a %s/create.yaml file for testing a Kubernetes controller.
```

Comment:
Please define the prompt as a const just before the function which uses it. (Eg https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/master/experiments/conductor/cmd/runner/fuzzer_commands.go#L68) It keeps the code more readable.